flict/flictlib/project/reader.py

```diff
@@ -0,0 +1,228 @@
+    """
+
+    def __init__(self, project_dirs):
+        self.spdx_dirs = project_dirs
+        self.files_read = []
+
+    def read_project(self, project_file):
+        packages_project_name = self._read_spdx(project_file)
+        packages = packages_project_name['packages']
+        project_name = packages_project_name['project_name']
+        flat_packages = self._flatten_packages(packages)
+
+        return {
+            "project_name": project_name,
+            "packages": flat_packages
+        }
+
+    def _read_spdx_2_2(self, only_packages=None):
+        project_name = self.project['name']
+        packages = {}
+        for pkg in self.project['packages']:
+            elem_id = pkg['SPDXID']
+            packages[elem_id] = {
+                'id': elem_id,
+                'name': pkg['name'],
+                'version': pkg['versionInfo'],
+                'license': pkg['licenseConcluded'],
+                'description': pkg['description'],
+                'dependencies': []
+            }
+
+        if 'relationships' in self.project:
+            for dep in self.project['relationships']:
+                top_package = dep['relatedSpdxElement']
+                dep_package_doc = dep['spdxElementId'].split(":")[0]
+                dep_package_name = dep['spdxElementId'].split(":")[1]
+                dep_spdx = dep_package_doc.replace("DocumentRef-", "") + ".spdx.json"
+                if self._already_read(dep_spdx):
+                    continue
+                self._add_files_read(dep_spdx)
+                dep_spdx_path = self.spdx_dirs[0] + "/" + dep_spdx
+                packages_proj_name = self._read_spdx(dep_spdx_path, self.spdx_dirs)
+                _packages = packages_proj_name.get('packages')
+                for _pkg in _packages.items():
+                    _pkg_name = _pkg[1]['id']
+                    if dep_package_name == _pkg_name:
+                        packages[top_package]['dependencies'].append({_pkg_name: _pkg[1]})
+                    else:
+                        pass
+
+        ret = {
+            "packages": packages,
+            "project_name": project_name
+        }
+        return ret
+
+    def _read_spdx(self, spdx_file, only_packages=None):
+
+        logging.info("_read_spdx: " + spdx_file)
+        with open(spdx_file, 'r') as f:
+            filename, suff = os.path.splitext(spdx_file)
+
+            if suff.lower() == ".json":
+                manifest_data = json.load(f)
+            else:
+                logging.error("Uknown format in file: " + spdx_file)
+                return None
+            logging.debug(" data read")
+
+        self.project = manifest_data
+        spdx_version = self.project['spdxVersion'].replace("SPDX-", "")
+
+        if spdx_version.startswith("2.2"):
+            return self._read_spdx_2_2(only_packages=None)
+        # TODO: raise exception with meaningful hint that SPDX version not supported
```

Comment:
pls fix that here - maybe with a generic FlictError, but returning with nothing where one expects a return is nothing we should merge like this